src/Bridges/ApplicationLatte/TemplateFactory.php

```diff
@@ -83,6 +83,12 @@ public function createTemplate(UI\Control $control = NULL)
 		$latte->addFilter('modifyDate', function ($time, $delta, $unit = NULL) {
 			return $time == NULL ? NULL : Nette\Utils\DateTime::from($time)->modify($delta . $unit); // intentionally ==
 		});
+		$latte->addFilter('translate', function ($text/*, ...$params*/) use ($template) {
+			if (($translator = $template->getTranslator()) && $translator instanceof Nette\Localization\ITranslator) {
+				return call_user_func_array([$translator, 'translate'], func_get_args());
+			}
+			throw new Nette\InvalidStateException('Translator has not been set.');
+		});
```

Comment:
What about this?
```php
if (!isset($latte->getFilters()['translate'])) {
    $latte->addFilter('translate', function () {
        throw new Nette\InvalidStateException('Translator has not been set.');
    });
}
```
and keep `..->addFilter` in the `Template::setTranslator` so it can overwrite the translate filter.